File: Models/Donor.cs
using System.ComponentModel.DataAnnotations;

namespace ChineseAuctionAPI.Models


{
    public class Donor
    {
        public int Id { get; set; }

        [MaxLength(100),Required(ErrorMessage = "First name is required")]
        public string FirstName { get; set; }

        [MaxLength(100), Required(ErrorMessage = "Last name is required")]
        public string LastName { get; set; }

        [MaxLength(100)]
        public string? Company { get; set; }

        public string Address { get; set; }

        [EmailAddress, Required(ErrorMessage = "Email is required")]
        public string Email { get; set; }

        [Phone]
        public string PhoneNumber { get; set; }
        public List<Prize> Prizes { get; set; } = new List<Prize>();
    }
}

File: client/models/Donor.ts
export interface DonorReadDTO {
  id: number;
  firstName?: string;
  lastName?: string;
  company?: string
  address?: string;
  email?: string;
  phoneNumber?: string;
  prizes?: ReadPrizeForDonorsDTO[];
}

export interface DonorCreateDTO {
  firstName?: string;
  lastName?: string;
  company?: string
  address?: string;
  email?: string;
  phoneNumber?: string;
}

export interface DonorUpdateDTO {
  id: number
  firstName?: string;
  lastName?: string;
  company?: string
  address?: string;
  email?: string;
  phoneNumber?: string;
}


export interface DonorForReadPrizesDTO {
  id: number;
  firstName?: string;
  lastName?: string;
  company?: string
  email?: string;
}

export interface ReadPrizeForDonorsDTO {
  id: number;
  name?: string;
  description?: string;
  categoryName?: string;
  imagePath?: string;
}